server/controllers/finance/reports/clientsReport/index.js

```diff
@@ -226,4 +223,176 @@ function escapeItems(list) {
   }, []);
 }
 
+/**
+ * @function getClientsData
+ * @param {object} options The object for filtering data using filterParser lib
+ * @param {boolean} isFromPeriodTotal is true if the data will be fetched from the period_total
+ * @param {boolean} isIniial is true if the data will be considered as initial 
+ * @description
+ * This function is responsible of returning a list of debtor and their 
+ * debit, credit, balance
+ **/
+function getClientsData(options, isFromPeriodTotal, isIniial) {
+  let finalQuery;
+  let parameters;
+
+  if (isFromPeriodTotal) {
```

Comment:
This function looks like this:
```js
function getClientsData(options, isFromPeriodTotal, isIniial) {
  if (isFromPeriodTotal) {
    // do something
  } else {
   // do something else
  }
}
```

It would probably be better to just make a series of unique functions, since the only repeating code is the `db.exec()` call.  Probably something like:
```js

// computes the client balances using the period total's table
getClientBalancesFromPeriodTotal()

getClientBalancesFromGeneralLedger()
```

This would make the code easier to read by:
 1. Removing an `if/else` block, so that the programmer doesn't have to remember too many logical layers of code
 2. Make the name of the function more descriptive to what the data source is
 3. `isIniial` only applies to the `isFromPeriodTotal` branch; the parameter is never used when `isFromPeriodTotal` is false.  Since we don't need it, we can only include it as a parameter in a function when it is needed.